### PR TITLE
Expose dapp interaction serialization methods and bump sargon version

### PR DESCRIPTION
**Pull request by jakub-rdx**

### CODE REVIEW

**Review by sergiupuhalschi-rdx — APPROVED** [review]

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import com.radixdlt.sargon.DappToWalletInteractionUnvalidated
+import com.radixdlt.sargon.dappToWalletInteractionUnvalidatedToJsonBytes
+import com.radixdlt.sargon.newDappToWalletInteractionUnvalidatedFromJsonBytes
+
+fun DappToWalletInteractionUnvalidated.Companion.fromJson(json: String) = runCatching {
+    newDappToWalletInteractionUnvalidatedFromJsonBytes(json.toByteArray().toBagOfBytes())
+}
+fun DappToWalletInteractionUnvalidated.toJson() = runCatching {
+    dappToWalletInteractionUnvalidatedToJsonBytes(this).string
+}
```
Comment (micbakos-rdx) [inline jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt:12]: We never exposed `Result` from kotlin extensions. We used to handle results in android and annotate these methods with `@Throws`. I would suggest keeping the same code style. Either expose result from all sargon methods or not.

File: jvm/sargon-desktop-debug/bin/main/darwin-aarch64/libsargon.dylib
Comment (micbakos-rdx) [inline jvm/sargon-desktop-debug/bin/main/darwin-aarch64/libsargon.dylib:1]: Why are these staged and committed? These build files that are gitignored. Please remove them

File: jvm/sargon-desktop-release/bin/main/darwin-aarch64/libsargon.dylib
Comment (micbakos-rdx) [inline jvm/sargon-desktop-release/bin/main/darwin-aarch64/libsargon.dylib:1]: The same for these

**Review by micbakos-rdx — CHANGES_REQUESTED** [review]
In general the conventions followed so far:

1. Methods that throw are annotated with @Throws(SargonException::class) and do not return result from sargon.
2. Make sure that methods indeed return a result in sargon and don't wrap everything under `runCatching` those. If something doesn't return a `rust Result` then it shouldn't throw.
3. Please remove committed build files.

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.radixdlt.sargon.DappToWalletInteractionUnvalidated
+import com.radixdlt.sargon.dappToWalletInteractionUnvalidatedToJsonBytes
+import com.radixdlt.sargon.newDappToWalletInteractionUnvalidatedFromJsonBytes
+
+@Throws(SargonException::class)
+fun DappToWalletInteractionUnvalidated.Companion.fromJson(json: String) =
+    newDappToWalletInteractionUnvalidatedFromJsonBytes(json.toByteArray().toBagOfBytes())
+
+@Throws(SargonException::class)
+fun DappToWalletInteractionUnvalidated.toJson() =
```
Comment (micbakos-rdx) [inline jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt:12]: Does this method throw an exception @CyonAlexRDX ?

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import com.radixdlt.sargon.newWalletToDappInteractionResponseFromJsonBytes
+import com.radixdlt.sargon.walletToDappInteractionResponseToJsonBytes
+
+@Throws(SargonException::class)
+fun WalletToDappInteractionResponse.Companion.fromJson(json: String) =
+    newWalletToDappInteractionResponseFromJsonBytes(json.toByteArray().toBagOfBytes())
+
+@Throws(SargonException::class)
+fun WalletToDappInteractionResponse.toJson() =
+    walletToDappInteractionResponseToJsonBytes(this).string
```
Comment (micbakos-rdx) [inline jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt:13]: Also does this method throw @CyonAlexRDX ?

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt (hunk 2 above)
Comment (CyonAlexRDX) [reply jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt:12]: No

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt (hunk 3 above)
Comment (CyonAlexRDX) [reply jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt:13]: To JSON never throws

Comment (CyonAlexRDX) [reply jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt:13]: @jakub-rdx only mark Kotlin as throwing if the Rust code returns `Result`

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt (hunk 2 above)
Comment (micbakos-rdx) [reply jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt:12]: I think @jakub-rdx if you open the generated `sargon.kt` you will see that this method does not throw. Neither this one https://github.com/radixdlt/sargon/pull/155/files#r1625795145

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt (hunk 3 above)
Comment (jakub-rdx) [reply jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt:13]: actually this one does not, at least it is not annotated in sargon.kt

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt (hunk 2 above)
Comment (jakub-rdx) [reply jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/DappToWalletInteractionUnvalidated.kt:11]: updated

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt (hunk 3 above)
Comment (jakub-rdx) [reply jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/WalletToDappInteractionResponse.kt:12]: @micbakos-rdx updated

**Review by micbakos-rdx — APPROVED** [review]

**Pull request merged**